templates/patient_extension.rb.erb

```diff
@@ -0,0 +1,137 @@
+  field :ethnicity, type: Code
+  field :race, type: Code
+  field :sex, type: Code
+
+  # These are the "data criteria", or QDM datatype elements that exist on a
+  # patient.
+  embeds_many :history_elements
+
+  # Returns an array of history elements that exist on this patient, that
+  # match the given HQMF data criteria OID.
+  def get_by_hqmf_oid(hqmf_oid)
+    history_elements.where(hqmf_oid: hqmf_oid) || []
+  end
+
+  # Returns an array of history elements that exist on this patient. Optionally
+  # takes a category, which returns all history elements of that QDM category.
+  # Example: patient.get_history_elements(category = 'encounters') will return
+  # all Encounter QDM data types active on the patient.
+  def get_history_elements(category = nil)
+    if type
+      history_elements.where(category: category) || []
+    else
+      history_elements || []
+    end
+  end
+
+  def adverse_events
+    get_history_elements('adverse_event')
+  end
+
+  def assessments
+    get_history_elements('assessment')
+  end
+
+  def allergies
+    get_history_elements('allergy')
+  end
+
+  def care_goals
+    get_history_elements('care_goal')
+  end
+
+  def conditions
+    get_history_elements('condition')
+  end
+
+  def encounters
+    get_history_elements('encounter')
+  end
+
+  def communications
+    get_history_elements('communication')
+  end
+
+  def family_history
+    get_history_elements('family_history')
+  end
+
+  def immunizations
+    get_history_elements('immunization')
+  end
+
+  def medical_equipment
+    get_history_elements('medical_equipment')
+  end
+
+  def medications
+    get_history_elements('medication')
+  end
+
+  def procedures
+    get_history_elements('procedure')
+  end
+
+  def results
+    get_history_elements('result')
+  end
+
+  def social_history
+    get_history_elements('social_history')
+  end
+
+  def vital_signs
+    get_history_elements('vital_sign')
+  end
+
+  def devices
+    get_history_elements('device')
```

Comment:
These should be re-sorted into alphabetical order by category type.  Looks like you are missing the 'symptom' type here.

Reply:
Good suggestion. And yes indeed, will need to add `symptom`.